### PR TITLE
Add Wanhao D12-300 profile

### PR DESCRIPTION
**Pull request by Musicdad12**

### CODE REVIEW

**Comment by SoftFever** [comment]
@Musicdad12 
Thank you for the PR.
The profile looks fine overall except two issues.
1. The bed model is not aligned. 
2. The bed texture's resolution is too low

The bed model and texture are optional, I would recommend leave them blank if there's not proper files.

File: resources/profiles/Wanhao/machine/Wanhao D12-300.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+    "type": "machine_model",
+    "name": "Wanhao D12/300",
+    "model_id": "D12",
+    "nozzle_diameter": "0.4",
+    "machine_tech": "FFF",
+    "family": "Wanhao",
+    "bed_model": "Wanhao D12-300_buildplate_model.stl",
```
Comment (SoftFever) [inline resources/profiles/Wanhao/machine/Wanhao D12-300.json:8]: We can leave bed_model and bed_texture blank if there is not proper files

**Review by Musicdad12 — COMMENTED** [review]
Fixed

**Review by SoftFever — APPROVED** [review]
Thanks!

**Pull request merged**